.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,102 @@
+---
+
+name: Build and Test
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+env:
+  IMAGE_PREFIX: "ghcr.io/${{ github.repository }}"
+  REGISTRY: 'ghcr.io'
+
+jobs:
+  build-test-push:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
• The pipeline should build these docker images in parallel. If any docker image fails to 
build successfully, then the pipeline should fail.